cms/grading/Sandbox.py

```diff
@@ -1215,8 +1201,38 @@ def _popen(self, command,
         return (Popen): popen object.
 
         """
-        self.exec_num += 1
         self.log = None
+        self.exec_num += 1
+
+        # We run a selection of commands without isolate, as they need
+        # to create new files. This is safe because these commands do
+        # not depend on the user input.
+        if command[0] in IsolateSandbox.SECURE_COMMANDS:
+            logger.debug("Executing non-securely: %s at %s",
+                         pretty_print_cmdline(command), self.path)
+            try:
+                prev_permissions = stat.S_IMODE(os.stat(self.path).st_mode)
+                os.chmod(self.path, 0777)
```

Comment:
Not really, everything is inside the mkdtemp which is created as 700 (you're also free to set a root dir for temporary files which has restrictions, instead of /tmp). Anyhow, since it actually works with less permission, I'll change to something less scary.